### PR TITLE
[Workers] Source Map beta

### PR DESCRIPTION
**Pull request by tanushree-sharma**
Follow ups before open beta: 

✅ add command to wrangler.toml config page: https://developers.cloudflare.com/workers/wrangler/configuration/
- add exception to example event https://developers.cloudflare.com/workers/observability/logging/tail-workers/#configure-tail-workers

### CODE REVIEW

File: content/workers/observability/source-maps.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+[Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. They show you the specific functions that were called, in what order, from which line and file, and with what arguments.
+
+Most JavaScript code – not just on Workers, but across platforms – is first bundled, often transpiled, and then minified before being deployed to production. This to create smaller bundles to optimize performance and convert from Typescript to Javascript if needed.
+
+Source maps translate compiled and minified code back to the original code that you wrote. Source maps are combined with the stack trace returned by the JavaScript runtime in order to present you with a stack trace.
+
+{{<Aside type="warning">}}
+
+Source maps and stack traces support is going to be in open beta on April 15.
```
Comment (irvinebroque) [inline content/workers/observability/source-maps.md]:
```suggestion
Source maps and stack traces support will be available in open beta starting on April 15. 
```

File: content/workers/observability/source-maps.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+{{<Aside type="warning">}}
+
+Source maps and stack traces support is going to be in open beta on April 15. 
+
+{{</Aside>}}
+
+## Source Maps
+
+Source maps can be enabled on a per-Worker level by adding the following to the Worker's [wrangler.toml](/workers/wrangler/) file:
```
Comment (irvinebroque) [inline content/workers/observability/source-maps.md]:
```suggestion
Add the following to your Worker's [wrangler.toml](/workers/wrangler/) file to enable source maps: 
```

File: content/workers/observability/source-maps.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+## Source Maps
+
+Source maps can be enabled on a per-Worker level by adding the following to the Worker's [wrangler.toml](/workers/wrangler/) file: 
+
+```toml
+upload_source_maps = true 
+
+```
+
+Once this setting is configured, wrangler will automatically generate and upload any source map files when you run [`wrangler deploy`](/workers/wrangler/commands/#deploy) or [`wrangler versions deploy`](/workers/wrangler/commands/#deploy-2).
```
Comment (irvinebroque) [inline content/workers/observability/source-maps.md]:
```suggestion
When `upload_source_maps` is set to `true`, Wrangler will automatically generate and upload source map files when you run [`wrangler deploy`](/workers/wrangler/commands/#deploy) or [`wrangler versions deploy`](/workers/wrangler/commands/#deploy-2).
```

File: content/workers/observability/source-maps.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+upload_source_maps = true 
+
+```
+
+Once this setting is configured, wrangler will automatically generate and upload any source map files when you run [`wrangler deploy`](/workers/wrangler/commands/#deploy) or [`wrangler versions deploy`](/workers/wrangler/commands/#deploy-2). 
+​​
+
+{{<Aside type="note">}}
+
+Miniflare supports [uploading source maps for local development](https://miniflare.dev/developing/source-maps).
```
Comment (irvinebroque) [inline content/workers/observability/source-maps.md]:
```suggestion
Miniflare can also [output source maps](https://miniflare.dev/developing/source-maps) for use local development or [testing](/workers/testing/integration-testing/#miniflares-api). 
```

File: content/workers/observability/source-maps.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+{{<Aside type="note">}}
+
+Miniflare supports [uploading source maps for local development](https://miniflare.dev/developing/source-maps). 
+
+{{</Aside>}}
+
+## Stack Traces
+​​
+When your Worker throws an uncaught exception, we fetch the source map and use it to map the stack trace of the exception back to lines of your Worker’s original source code.
```
Comment (irvinebroque) [inline content/workers/observability/source-maps.md]:
```suggestion
When your Worker throws an uncaught exception, after your Worker invocation completes, the Workers runtime will fetch the source map file that you uploaded and use it to map the stack trace of the exception back to lines of your Worker’s original source code.
```

Can rephrase if you want but think important to be clear that you won't get a sourcemapped stack trace for an error that is exposed at runtime in your own Worker

**Review by irvinebroque — APPROVED** [review]

File: content/workers/observability/source-maps.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+---
+pcx_content_type: concept
```
Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
pcx_content_type: configuration
```

File: content/workers/observability/source-maps.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+title: Source maps and stack traces
+meta:
+  description: Adding source maps and generating stack traces for Workers. 
+---
+{{<heading-pill style="beta">}}Source maps and stack traces{{</heading-pill>}}
+
+[Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. They show you the specific functions that were called, in what order, from which line and file, and with what arguments.
```
Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
[Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. Stack traces show you the specific functions that were called, in what order, from which line and file, and with what arguments.
```

Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
[Stack traces](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/stack) help with debugging your code when the Worker encounters an unhandled exception. Stack traces show you the specific functions that were called, in what order, from which line and file, and with what arguments.
```

File: content/workers/observability/source-maps.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+Most JavaScript code – not just on Workers, but across platforms – is first bundled, often transpiled, and then minified before being deployed to production. This to create smaller bundles to optimize performance and convert from Typescript to Javascript if needed.
```
Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
Most JavaScript code – not just on Workers, but across platforms – is first bundled, often transpiled, and then minified before being deployed to production. This process creates smaller bundles to optimize performance and converts code from Typescript to Javascript if needed.
```

File: content/workers/observability/source-maps.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+Source maps translate compiled and minified code back to the original code that you wrote. Source maps are combined with the stack trace returned by the JavaScript runtime in order to present you with a stack trace.
```
Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
Source maps translate compiled and minified code back to the original code that you wrote. Source maps are combined with the stack trace returned by the JavaScript runtime to present you with a stack trace.
```

File: content/workers/observability/source-maps.md (hunk 2 above)
Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
To enable source maps, add the following to your Worker's [`wrangler.toml`](/workers/wrangler/configuration/) file: 
```

File: content/workers/observability/source-maps.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+
+{{</Aside>}}
+
+## Source Maps
+
+Source maps can be enabled on a per-Worker level by adding the following to the Worker's [wrangler.toml](/workers/wrangler/) file: 
+
+```toml
+upload_source_maps = true 
+
```
Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
```

File: content/workers/observability/source-maps.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+Once this setting is configured, wrangler will automatically generate and upload any source map files when you run [`wrangler deploy`](/workers/wrangler/commands/#deploy) or [`wrangler versions deploy`](/workers/wrangler/commands/#deploy-2). 
+​​
+
+{{<Aside type="note">}}
+
+Miniflare supports [uploading source maps for local development](https://miniflare.dev/developing/source-maps). 
+
+{{</Aside>}}
+
+## Stack Traces
```
Comment (deadlypants1973) [inline content/workers/observability/source-maps.md]:
```suggestion
## Stack traces
```

**Review by deadlypants1973 — APPROVED** [review]

**Comment by deadlypants1973** [comment]
@tanushreesharma-cf I would put a link to this page under ## Related resources in Tail Workers and vice verse for Tail Workers in Stack traces ## Related resources

File: content/workers/observability/source-maps.md (hunk 5 above)
Comment (tanushree-sharma) [reply content/workers/observability/source-maps.md]: Yep, agreed. Adding

**Pull request merged**